app/services/DFS.coffee

```diff
@@ -0,0 +1,78 @@
+Tuple    = require "../models/Tuple"
+TupleSet = require "../models/TupleSet"
+
+class DFS
+
+  @isValidSeed: (x, y) ->
+    if x - 1 < 0 or x + 1 >= @grid.dimension
+      return false
+    colCount = []
+    for col in [x - 1, x, x + 1]
+      count = 0
+      for row in [0...@grid.dimension]
+        if @grid.grid[row][col] is " "
+          count += 1
+      colCount.push count
+    if colCount[1] is 1 and colCount[0] > 0 and colCount[2] > 0
+      return true
+    false
+
+  @shuffle: (array) ->
+    # Fisher-Yates shuffle
+    m = array.length
+    while m
+      i = Math.floor(Math.random() * m--)
+      t = array[m]
+      array[m] = array[i]
+      array[i] = t
+    array
+
+  @search: (seed, input, takenCells) ->
+    return true if input.length is 0
+
+    toVisit = new @AdjacentCells @grid, null, seed.x, seed.y
+    toVisit = @shuffle toVisit.calculate takenCells.list
```

Comment:
This scares me.   Its technically okay, but my static-typed background wants pointers to only reference one type
```
calculator = new @AdjacentCells @grid, null, seed.x, seed.y
toVisit = @shuffle calculator.calculate takenCells.list
```